client/dashboard/sites/overview-plan-card/index.tsx

```diff
@@ -33,15 +117,89 @@ function getCurrentMonthRangeTimestamps() {
 	};
 }
 
-export default function PlanCard( { site }: { site: Site } ) {
-	const { data: plan, isLoading: isLoadingPlan } = useQuery( siteCurrentPlanQuery( site.ID ) );
-	const { data: purchase, isLoading: isLoadingPurchase } = useQuery( {
-		...sitePurchaseQuery( site.ID, plan?.id ?? '' ),
-		enabled: !! plan?.id,
-	} );
+function getJetpackProductsDescription( products: typeof JETPACK_PRODUCTS ) {
+	if ( products.length === JETPACK_PRODUCTS.length ) {
+		return __( 'The full Jetpack suite with everything you need to grow your business.' );
+	}
+
+	if ( products.length === 0 ) {
+		return __( 'Enhance your site with Jetpack security, performance, and growth tools.' );
+	}
+
+	if ( products.length === 1 ) {
+		return products[ 0 ].description;
+	}
+
+	return `${ products.map( ( product ) => product.label ).join( ', ' ) }.`;
+}
+
+function getJetpackProducts( site: Site ) {
+	return JETPACK_PRODUCTS.filter( ( product ) =>
+		hasPlanFeature( site, product.id as JetpackFeatures )
+	);
+}
+
+function JetpackPlanCard( {
+	site,
+	purchase,
+	isLoading,
+}: {
+	site: Site;
+	purchase?: Purchase;
+	isLoading: boolean;
+} ) {
+	const products = getJetpackProducts( site );
+	const hasProducts = products.length > 0;
+	const productsToDisplay = hasProducts ? products : JETPACK_PRODUCTS;
+	const isPlanFreeAndHasProducts = site.plan?.is_free && hasProducts;
+
+	return (
+		<OverviewCard
+			title={ __( 'Subscriptions' ) }
+			icon={ <JetpackLogo /> }
+			heading={ isPlanFreeAndHasProducts ? __( 'Jetpack' ) : site.plan?.product_name_short }
```

Comment:
Shouldn't we check if `products.length === 1` here, in which case, we should show that product?

Currently, if I have only Jetpack Scan for example, it shows Jetpack; we should be showing Jetpack Scan instead:

<img width="332" height="321" alt="Image" src="https://github.com/user-attachments/assets/d38fcd8a-c23f-4e39-8847-891512962a1f" />

Reply:
Updated the logic to the following:

1. If Jetpack Free
  a. If one module, then show module name
  b. If more than one, show "Jetpack"
2. Otherwise, show the full plan name

Reply:
I'm now wondering if we should bring this logic to the DataViews Plan field... @matt-west thoughts?

Reply:
Yep, makes sense for the plan card and the column in the sites table to be consistent.